### PR TITLE
Feature/devise

### PR DESCRIPTION
**Pull request by flemton**
## Dear reviewer, it is a pleasure to request a review of my devise (auth) feature before merging into dev. 
Below were the changes made 
- ### Remove current_user method in ApplicationController
- ### [Install and setup devise](https://github.com/heartcombo/devise#getting-started).
      Can register a new user.
      User logs in with a combination of email and password.
      Hashed password stored in the database.
      Ask for confirmation of email.
      Can reset password.

### CODE REVIEW

**Review by abdulhamiid — COMMENTED** [review]
Hi @flemton 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request ❌ ❌ ❌ 

- [ ] This milestone is a [pair-programming activity](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_devise.md#work-and-submission-mode). We will check the commit history to make sure that everybody has contributed code on each project. 🛠

- [ ] Please ensure that everyone has contributed equally to the project you're reviewing. 😊

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by flemton** [comment]
Thank you very much sir @abdulhamiid , cheers 🥂

File: db/schema.rb [hunk 1]
```diff
@@ -52,6 +52,16 @@
...
+    t.text "bio"
+    t.string "confirmation_token"
+    t.datetime "confirmed_at"
+    t.datetime "confirmation_sent_at"
+    t.string "unconfirmed_email"
+    t.string "encrypted_password"
+    t.datetime "remember_created_at"
+    t.string "reset_password_token"
+    t.datetime "reset_password_sent_at"
+    t.index ["confirmation_token"], name: "index_users_on_confirmation_token", unique: true
```
Comment (tugsem) [inline db/schema.rb]: - [ ] I think you have misunderstood the devise setup. These columns including 'email' will be added by the migration named 'add_devise_to_users' which we added through this command 'rails g devise user'. But it seems like you have removed the necessary lines from that file and added these columns yourself. Please remove these migration files first and then rerun 'rails g devise user' and uncomment the 'confirmable' section. You need to remove 'email' column first from users to not face an error. Kindly implement this.

**Review by tugsem — CHANGES_REQUESTED** [review]
##  STATUS: REQUIRED CHANGES ♻️
Hello @flemton  and @diarisdiakite 👋 , 

Good job so far! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text
## Highlights 🚀 
✔️linters are passing
✔️correct use of gitflow
✔️almost done with the devise setup
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/schema.rb (hunk 1 above)
Comment (flemton) [reply db/schema.rb]: Dear @tugsem, thank you very much for the review and suggestions, the suggested changes are kindly implemented

**Review by KayLemba — APPROVED** [review]
# STATUS: APPROVED 🟢 
Hello @flemton   ,
Your project is complete! There is nothing else to say other than... it's time to merge it  .
Congratulations! 🎉 Keep up the Awesome work 🥂 Keep Rocking 🥇

HIGHLIGHTS:

- [x]  Your repo is professional 😎 

- [x] Descriptive P.R 💯 Professional README file

- [x] App runs as intended

- [x] Nice UI Design

Now you can merge this branch and move on 🥇
Happy Coding cheers 🥇 💯

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using

**Pull request merged**